PowerPointLabs/PowerPointLabs/SyncLab/SyncLabShapeStorage.cs

```diff
@@ -6,22 +6,19 @@
 
 namespace PowerPointLabs.SyncLab
 {
-    public class SyncLabShapeStorage : PowerPointPresentation
+    public sealed class SyncLabShapeStorage : PowerPointPresentation
     {
 
         public const int FormatStorageSlide = 0;
 
         private int nextKey = 0;
 
-        private static SyncLabShapeStorage storage;
+        private static readonly Lazy<SyncLabShapeStorage> StorageInstance =
```

Comment:
Lazy Type will only create the instance once when it is needed. Subsequently, it will not be created. In a sense, it is a built in singleton with thread safe.